src/drivers/imx/esai.c

```diff
@@ -4,37 +4,456 @@
+		tracev_esai("ESAI: Bit clock NORMAL frame INVERTED");
+		/* Invert frame clock */
+		xccr ^= ESAI_xCCR_xFSP;
+		break;
+	case SOF_DAI_FMT_IB_NF:
+		tracev_esai("ESAI: Bit clock INVERTED frame NORMAL");
+		/* Invert bit clock */
+		xccr ^= ESAI_xCCR_xCKP | ESAI_xCCR_xHCKP;
+		break;
+	case SOF_DAI_FMT_IB_IF:
+		tracev_esai("ESAI: Bit clock INVERTED frame INVERTED");
+		/* Invert both clocks */
+		xccr ^= ESAI_xCCR_xCKP | ESAI_xCCR_xHCKP | ESAI_xCCR_xFSP;
+		break;
+	default:
+		trace_esai_error("ESAI: Invalid bit inversion format");
+		return -EINVAL;
+	}
+
+	switch (config->format & SOF_DAI_FMT_MASTER_MASK) {
+	case SOF_DAI_FMT_CBM_CFM:
+		tracev_esai("ESAI: Code clock MASTER codec frame MASTER");
+		/* Nothing to do in the registers */
+		break;
+	case SOF_DAI_FMT_CBM_CFS:
+		tracev_esai("ESAI: Code clock MASTER codec frame SLAVE");
+		xccr |= ESAI_xCCR_xFSD;
+		break;
+	case SOF_DAI_FMT_CBS_CFM:
+		tracev_esai("ESAI: Code clock SLAVE codec frame MASTER");
+		xccr |= ESAI_xCCR_xCKD;
+		break;
+	case SOF_DAI_FMT_CBS_CFS:
+		tracev_esai("ESAI: Code clock SLAVE codec frame SLAVE");
+		xccr |= ESAI_xCCR_xFSD | ESAI_xCCR_xCKD;
+		break;
+	default:
+		trace_esai_error("ESAI: Invalid master-slave configuration");
+		return -EINVAL;
+	}
+
+	/* Set networked mode; we only support 2 channels now, not 1 */
+	xcr |= ESAI_xCR_xMOD_NETWORK;
+	xccr |= ESAI_xCCR_xDC(2);
+
+	/* Word size -- TODO compute correct one */
+	xcr |= ESAI_xCR_xSWS(32, 24) | ESAI_xCR_PADC;
+
+	/* Remove "RESET" flag */
+	dai_update_bits(dai, REG_ESAI_ECR, ESAI_ECR_ERST, 0);
+	
+	dai_update_bits(dai, REG_ESAI_ECR, ESAI_ECR_ETI, ESAI_ECR_ETI);
+
+	mask = ESAI_xCCR_xCKP | ESAI_xCCR_xHCKP | ESAI_xCCR_xFSP |
+		ESAI_xCCR_xFSD | ESAI_xCCR_xCKD | ESAI_xCCR_xHCKD |
+		ESAI_xCCR_xDC_MASK;
+
+	xccr |= ESAI_xCCR_xHCKD; /* Set the HCKT pin as an output */
+
+	if (~mask & xccr) {
+		trace_esai_error("XCCR bits not caught by mask: 0x%08x",
+				 (~mask & xccr));
+		trace_esai_error("MASK 0x%08x XCCR 0x%08x", mask, xccr);
+	}
+
+	dai_update_bits(dai, REG_ESAI_TCCR, mask, xccr); /* rx */
+	dai_update_bits(dai, REG_ESAI_RCCR, mask, xccr); /* tx */
+
+	mask = ESAI_xCR_xFSL | ESAI_xCR_xFSR | ESAI_xCR_xWA |
+		ESAI_xCR_xMOD_MASK | ESAI_xCR_xSWS_MASK | ESAI_xCR_PADC |
+		ESAI_xCR_xPR;
+	xcr |= ESAI_xCR_xPR;
+
+	if (~mask & xcr) {
+		trace_esai_error("XCR bits not caught by mask: 0x%08x",
+				 (~mask & xcr));
+		trace_esai_error("MASK 0x%08x XCR 0x%08x", mask, xcr);
+	}
+
+	dai_update_bits(dai, REG_ESAI_TCR, mask, xcr); /* rx */
+	dai_update_bits(dai, REG_ESAI_RCR, mask, xcr); /* tx */
+
+	dai_write(dai, REG_ESAI_TSMA, 0);
+	dai_write(dai, REG_ESAI_TSMB, 0);
+	dai_write(dai, REG_ESAI_RSMA, 0);
+	dai_write(dai, REG_ESAI_RSMB, 0);
+
+	/* Program FIFOs */
+	dai_update_bits(dai, REG_ESAI_RFCR, ESAI_xFCR_xFR, ESAI_xFCR_xFR);
+	dai_update_bits(dai, REG_ESAI_RFCR, ESAI_xFCR_xFR, 0);
+	
+	/* TODO use TFCR and RFCR */
+	dai_update_bits(dai, REG_ESAI_TFCR,
+				ESAI_xFCR_xFR_MASK,
+				ESAI_xFCR_xFR);
```

Comment:
I think the second line isn't correctly aligned.

Reply:
Not sure what happened, fixed. Also checking if I have this issue in other places too.